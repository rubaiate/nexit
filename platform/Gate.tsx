import React from "react";
import { Rect } from "react-native-svg";
interface Props {
  x: number;
  y: number;
  width: number;
  height: number;
}

export const Gate = (props: Props) => {
  const { x, y, height, width } = props;
  return (
    <Rect
      x={x}
      y={y - height / 2}
      width={width}
      height={height}
      stroke="black"
      strokeWidth="1"
      fill="#173F5F"
    />
  );
};
