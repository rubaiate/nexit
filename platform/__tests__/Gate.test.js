import React from 'react';
import renderer from 'react-test-renderer';

import Gate from '../Gate'

it('renders correctly', () => {
    const tree = renderer.create(<Gate x={10} y={10} width={1000} height={60}/>).toJSON();
    expect(tree).toMatchSnapshot()
})