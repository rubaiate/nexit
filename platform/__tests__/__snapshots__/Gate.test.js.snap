// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNSVGRect
  fill={4279713631}
  height={60}
  propList={
    Array [
      "fill",
      "stroke",
      "strokeWidth",
    ]
  }
  stroke={4278190080}
  strokeWidth="1"
  width={1000}
  x={10}
  y={-20}
/>
`;
